src/random_lcg.cpp

```diff
@@ -28,47 +27,41 @@ constexpr uint64_t prn_stride {152917LL};                // stride between
                                                          //   particles
 constexpr double   prn_norm   {1.0 / prn_mod};           // 2^-63
 
-// Current PRNG state
-uint64_t prn_seed[N_STREAMS];  // current seed
-int      stream;               // current RNG stream
-#pragma omp threadprivate(prn_seed, stream)
-
-
 //==============================================================================
 // PRN
 //==============================================================================
 
 extern "C" double
-prn()
+prn(uint64_t* prn_seeds, int stream)
 {
   // This algorithm uses bit-masking to find the next integer(8) value to be
   // used to calculate the random number.
-  prn_seed[stream] = (prn_mult*prn_seed[stream] + prn_add) & prn_mask;
+  prn_seeds[stream] = (prn_mult*prn_seeds[stream] + prn_add) & prn_mask;
 
   // Once the integer is calculated, we just need to divide by 2**m,
   // represented here as multiplying by a pre-calculated factor
-  return prn_seed[stream] * prn_norm;
+  return prn_seeds[stream] * prn_norm;
 }
 
 //==============================================================================
 // FUTURE_PRN
 //==============================================================================
 
 extern "C" double
-future_prn(int64_t n)
+future_prn(int64_t n, uint64_t* prn_seeds, int stream)
```

Comment:
Same thought here regarding the interface. I think passing a single seed would suffice.